crates/fluvio/Cargo.toml

```diff
@@ -46,6 +46,7 @@ quantities = { version = "0.10.0", features = ["datavolume", "datathroughput", "
 hdrhistogram = { version = "7.5.0", optional = true }
 strum = { version = "0.24", features = ["derive"], optional = true }
 num-traits = { version = "0.2", optional = true }
+async-std = { version = "1.12", default-features = false, features=["std"] }
```

Comment:
Unfortunately, `async-std` bring large sets of dependencies.  Can you try to use https://docs.rs/event-listener?  `CondVar` uses internally.

Reply:
The key feature of `Condvar` that `Event Listener` doesn't have is that you can `wait` under the lock: it drops the lock, goes to sleep, and then regains the lock after awaking `automatically` unnoticeable for our code. With `Event Listener` all these manipulations need to be done manually.

Reply:
ok. make sense.  However, we should not bring rest of dependency from async-std which will make it harder for us to port to WASM or more strict env.  Can we just port `CondVar`?

Reply:
Looks like we are including `std` due to `task` feature flag in the `fluvio_future`.   Can we re-export `CondVar` in the `fluvio-future` under `sync` flag?  That way, we don't explicit depend on `async_std` directly from Fluvio?

Reply:
Done